src/main/java/com/speaker/service/util/StringParser.java

```diff
@@ -0,0 +1,31 @@
+package com.speaker.service.util;
+
+import java.util.Optional;
+
+import static java.util.Objects.nonNull;
+
+public class StringParser {
+    private static final Integer LIMIT_VALUE = 2;
+
+    public static Optional<Pair<String, String>> parseOptionalStringBySpace(String names) {
+        if (nonNull(names)) {
+            return parser(names);
+        }
+        return Optional.empty();
+    }
+
+    public static Optional<Pair<String, String>> parseOptionalStringBySpace(Optional<String> names) {
+        if (nonNull(names) && names.isPresent()) {
```

Comment:
again `nonNull(names)` whay?

Reply:
Comments accepted, corrections made. thank you. it is mistake